daemon/src/model.rs

```diff
@@ -635,32 +638,131 @@ impl Default for FundingRate {
     }
 }
 
-/// Fee paid on every contract renewal (a fraction of it if rolling over)
-#[derive(Debug, Clone)]
+/// Fee paid by the taker to the maker for opening a cfd.
+#[derive(Debug, Copy, Clone, PartialEq, PartialOrd, Serialize, Deserialize)]
+pub struct OpeningFee {
+    #[serde(with = "::bdk::bitcoin::util::amount::serde::as_btc")]
+    amount: Amount,
+}
+
+impl OpeningFee {
+    pub fn new(amount: Amount) -> Self {
+        Self { amount }
+    }
+
+    /// Value in satoshis for the taker
+    pub fn to_u64(&self) -> u64 {
+        self.amount.as_sat()
+    }
+
+    pub fn amount_for(&self, role: Role) -> SignedAmount {
+        let amount_sat: i64 = self.amount.as_sat().try_into().expect("to convert");
+        let amount_sat = match role {
+            Role::Maker => -amount_sat,
+            Role::Taker => amount_sat,
+        };
+        SignedAmount::from_sat(amount_sat)
+    }
+}
+
+impl fmt::Display for OpeningFee {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        self.amount.fmt(f)
```

Comment:
Did you test what this does? I think that will `Display` to something like `0 BTC`!

I think we should only store the satoshi value in the DB so you either need to adopt this implementation or do a manual implementation of what the `impl_sqlx_type_display_from_str` macro does.

Reply:
I'll store it as satoshis then?

Reply:
We don't have tests with anything but 0s here, so the tests would probably not catch it :)

Reply:
as spoken offline, changed to a newtype storing `u64` satoshis.